pkg/datapath/connector/ipvlan.go

```diff
@@ -30,10 +30,10 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-func getEntryProgInstructions(fd int) asm.Instructions {
+func getEntryProgInstructions(fd int, index int32) asm.Instructions {
```

Comment:
I'd introduce a type and two consts DirIngress and DirEgress for `index` instead of `int32`. That would improve readability and help to avoid mixing them up in the code.

Reply:
Yes, two consts (Ingress and Egress) for TailMap may be a good choice, I will add them in next commit. But for the function ```getEntryProgInstructions```, I think the ```int32``` type of ```index``` is appropriate. ```getEntryProgInstructions``` creates a bpf program that calls map[index], this ```index``` is only the index of map, not the direction.

Reply:
Sounds good to me.